src/socket_poller.cpp

```diff
@@ -464,117 +592,28 @@ int zmq::socket_poller_t::wait (zmq::socket_poller_t::event_t *events_, int n_ev
             signaler->recv ();
 
         //  Check for the events.
-        int found = 0;
-        for (items_t::iterator it = items.begin (); it != items.end () && found < n_events_; ++it) {
-
-            events_[found].socket = NULL;
-            events_[found].fd = 0;
-            events_[found].user_data = NULL;
-            events_[found].events = 0;
-
-            //  The poll item is a 0MQ socket. Retrieve pending events
-            //  using the ZMQ_EVENTS socket option.
-            if (it->socket) {
-                size_t events_size = sizeof (uint32_t);
-                uint32_t events;
-                if (it->socket->getsockopt (ZMQ_EVENTS, &events, &events_size) == -1) {
-                    return -1;
-                }
-
-                if (it->events & events) {
-                    events_[found].socket = it->socket;
-                    events_[found].user_data = it->user_data;
-                    events_[found].events = it->events & events;
-                    ++found;
-                }
-            }
-            //  Else, the poll item is a raw file descriptor, simply convert
-            //  the events to zmq_pollitem_t-style format.
-            else {
-                short revents = pollfds [it->pollfd_index].revents;
-                short events = 0;
-
-                if (revents & POLLIN)
-                    events |= ZMQ_POLLIN;
-                if (revents & POLLOUT)
-                    events |= ZMQ_POLLOUT;
-                if (revents & POLLPRI)
-                    events |= ZMQ_POLLPRI;
-                if (revents & ~(POLLIN | POLLOUT | POLLPRI))
-                    events |= ZMQ_POLLERR;
-
-                if (events) {
-                    events_[found].socket = NULL;
-                    events_[found].user_data = it->user_data;
-                    events_[found].fd = it->fd;
-                    events_[found].events = events;
-                    ++found;
-                }
-            }
-        }
+        int found = check_events(events_, n_events_);
         if (found) {
-            for (int i = found; i < n_events_; ++i) {
-                events_[i].socket = NULL;
-                events_[i].fd = 0;
-                events_[i].user_data = NULL;
-                events_[i].events = 0;
-            }
+            if (found > 0)
+                zero_trail_events(events_, n_events_, found);
```

Comment:
missing space before (

Reply:
Is there any `checkpatch.pl`-like which does style checkings on the patch before PRing? It is really annoying (sorry I contribute to many different projects with various styles)

Reply:
I think so too. I made a clang-format config file for myself, but clang-format does not support everything in the ZeroMQ style. In particular, it does not support adding spaces before indexing square brackets :( I would really appreciate if the style could be moved to something that can be expressed using clang-format, which is a very good tool IMO, and one of the few (the only one?) that works with a real C++ parser.